nixos/modules/services/web-apps/archtika.nix

```diff
@@ -0,0 +1,312 @@
+    enable = mkEnableOption "archtika service";
+
+    package = mkPackageOption pkgs "archtika" { };
+
+    user = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "User account under which archtika runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "Group under which archtika runs.";
+    };
+
+    databaseName = mkOption {
+      type = types.str;
+      default = "archtika";
+      description = "Name of the PostgreSQL database for archtika.";
+    };
+
+    apiPort = mkOption {
+      type = types.port;
+      default = 5000;
+      description = "Port on which the API runs.";
+    };
+
+    apiAdminPort = mkOption {
+      type = types.port;
+      default = 7500;
+      description = "Port on which the API admin server runs.";
+    };
+
+    webAppPort = mkOption {
+      type = types.port;
+      default = 10000;
+      description = "Port on which the web application runs.";
+    };
+
+    domain = mkOption {
+      type = types.str;
+      default = null;
+      description = "Domain to use for the application.";
+    };
+
+    acmeEmail = mkOption {
+      type = types.str;
+      default = null;
+      description = "Email to notify for the SSL certificate renewal process.";
+    };
+
+    dnsProvider = mkOption {
+      type = types.str;
+      default = null;
+      description = "DNS provider for the DNS-01 challenge (required for wildcard domains).";
+    };
+
+    dnsEnvironmentFile = mkOption {
+      type = types.path;
+      default = null;
+      description = "API secrets for the DNS-01 challenge (required for wildcard domains).";
+    };
+
+    settings = mkOption {
+      description = "Settings for the running archtika application.";
+      type = types.submodule {
+        options = {
+          disableRegistration = mkOption {
+            type = types.bool;
+            default = false;
+            description = "By default any user can create an account. That behavior can be disabled by using this option.";
+          };
+          maxUserWebsites = mkOption {
+            type = types.int;
+            default = 2;
+            description = "Maximum number of websites allowed per user by default.";
+          };
+          maxWebsiteStorageSize = mkOption {
+            type = types.int;
+            default = 500;
+            description = "Maximum amount of disk space in MB allowed per user website by default.";
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users.${cfg.user} = {
+      isSystemUser = true;
+      group = cfg.group;
+    };
+
+    users.groups.${cfg.group} = {
+      members = [
+        "nginx"
+        "postgres"
+      ];
+    };
+
+    systemd.tmpfiles.rules = [
+      "d /var/www 0755 root root -"
+      "d /var/www/archtika-websites 0770 ${cfg.user} ${cfg.group} -"
+    ];
+
+    systemd.services.archtika-api = {
```

Comment:
Does the package provide any systemd units, or are you expected to make your own when using Archtika?

Reply:
I may have misinterpreted your question, but when using the module, two hardened systemd services are automatically set up for the REST API and the web application, so the application will run long term.

Reply:
I meant if upstream included any generic systemd units in `lib/systemd` when compiling the package. Those can be used instead of making whole new ones.

Reply:
Oh okay, no, that is not the case, which is why the units had to be explicitly declared in the module.

Reply:
A shame. Feel free to mark this as resolved then.